### DISCUSSION TITLE
Play audio via go2rtc

### DISCUSSION THREAD

**Discussion by Matzebhv** [blakeblackshear/frigate#6145] [General]
Hey guys,

what do i have to configure to enabling playing an mp3 stream or file to one of my cams via go2rtc?
Using the play audio link in the dashboard results in nothing, browser dev-console says -> Failed to load resource: the server responded with a status of 500 (Internal Server Error).

The cam supports 2way audio.

go2rtc:
  streams:
    Teich:
      - rtsp://192.168.178.238:554/11 #High Res
      - "ffmpeg:Teich#audio=opus"
    Teich_sub:
      rtsp://192.168.178.238:554/12 #Low Res
  webrtc:
    candidates:
      - 192.168.178.15:8555
      - stun:8555
    
Regards
Matthias

**Comment by NickM-27** [comment]
what audio file are you trying to play? most likely you are using a path that does not exist in the container

**Reply by Matzebhv** [reply]
Ahh, thats a good hint. Sorry, but i am a total docker noob at this time. I have tried an external mp3 stream.

Will try an local file.

**Comment by Matzebhv** [comment]
No, this does not work to.
Am i right, i have - /root/Documents/frigate:/media/frigate in my compose.yml. 
I have put an test.mp3 under /root/Documents/frigate,

Then putting /media/frigate/test.mp3 into play audio in the dashboard should work?

Regards
Matthias

**Reply by NickM-27** [reply]
yes that should work, most likely you have some other issue in what is being submitted on the webUI

**Comment by ehn** [comment]
Were you able to get this to work? I am similarly unable to. I have put a random audio file in `/media/frigate/test.mp3`. In my Docker compose config, I have `- /media/frigate:/media/frigate` under `volumes:`. Yet, the camera makes no sound. There is nothing in the go2rtc log.

However, in the NGINX log, there is this line, which seems to indicate a problem (HTTP status `500`):

```
2023-05-02 07:04:59.374877457  192.168.193.13 - - [02/May/2023:07:04:59 +0000] "POST /live/webrtc/api/streams?dst=test_camera&src=%2Fmedia%2Ffrigate%2Ftest.mp3 HTTP/1.1" 500 44 "http://192.168.1.1:5000/live/webrtc/links.html?src=test_camera" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36" "-"
```

The camera I'm trying this on is a [TP-Link Tapo C210](https://www.tp-link.com/sg/home-networking/cloud-camera/tapo-c210/).

**Reply by blakeblackshear** [reply]
Do the go2rtc logs show an error?

**Reply by ehn** [reply]
No, nothing.

**Comment by Matzebhv** [comment]
@ehn I had no time to dig deeper into this problem, no. In my case it does not work. If i find the time i will take a look with wireshark.

**Reply by NateMeyer** [reply]
I had some issue with the 2-way audio in the version of go2rtc included in frigate 0.12.  But I was able to get 2-way audio working via webrtc with version 1.3.2 and 1.4 of go2rtc running in a stand-alone container from frigate.  Not sure if the same issue applies to playing an audio file, but it could be worth trying.

**Reply by NickM-27** [reply]
To be clear you can also easily override the internal go2rtc with a newer version.

**Reply by Vladd1980** [reply]
did anyone manage to send the audio in the end?

**Reply by NickM-27** [reply]
works fine for me

**Reply by Vladd1980** [reply]
can you give an example of a ready-made command? I've been trying to implement this for two weeks now and nothing is working. unfortunately my level of knowledge is too low

**Reply by Vladd1980** [reply]
I'm trying to do this using this command from the go2rtc web interface. there is access to the file,
ffmpeg -re -i http://192.168.200.9:8123/local/111.mp3 -f rtsp rtsp://192.168.201.8:8554/Yard1 && curl -X POST "http://192.168.201.18/cgi-bin/audio_input" -d "source=rtsp://admin:*****@192.168.201.8:8554/Yard1"

**Reply by NickM-27** [reply]
you are not using go2rtc, what you're doing doesn't make sense. This is what works for me

`http://GO2RTC_IP:1984/api/streams?dst=NAME_OF_CAMERA_IN_GO2RTC&src=ffmpeg%3Ahttps%3A%2F%2Fquicksounds.com%2Fuploads%2Ftracks%2F1548821975_1522671138_1927006130.mp3%23audio%3Dpcma%23input%3Dfile`

you can easily achieve this by using the go2rtc dashboard -> select `links` on the camera of interest -> input `ffmpeg:https://quicksounds.com/uploads/tracks/1548821975_1522671138_1927006130.mp3#audio=pcma#input=file` into the field under **Play Audio** and press send

**Reply by Vladd1980** [reply]
![image](https://github.com/user-attachments/assets/6df3602b-bed0-4f7d-96d4-6b76815818c2)
ffmpeg:http://192.168.200.9:8123/local/111.mp3#audio=pcma#input=file
and again nothing happens :(

**Reply by NickM-27** [reply]
Then your camera isn't setup correctly or doesn't support the feature

**Reply by Vladd1980** [reply]
The function is supported. I send voice messages from the xmey application without any problems. What camera settings should you check?

**Reply by NickM-27** [reply]
Just because your camera supports it via app doesn't mean it supports the features to do it via go2rtc. You should read the go2rtc docs on the subject.
